src/core/net/ip6_mpl.cpp

```diff
@@ -164,14 +168,16 @@ void Mpl::AddBufferedMessage(Message &aMessage, uint16_t aSeedId, uint8_t aSeque
     Message *messageCopy = NULL;
     MplBufferedMessageMetadata messageMetadata;
     uint32_t nextTransmissionTime;
-    uint8_t hopLimit;
+    uint8_t hopLimit = 0;
+
+    otLogFuncEntry();
 
     VerifyOrExit((messageCopy = aMessage.Clone()) != NULL, error = kThreadError_NoBufs);
 
     if (!aIsOutbound)
     {
         aMessage.Read(Header::GetHopLimitOffset(), Header::GetHopLimitSize(), &hopLimit);
-        VerifyOrExit(--hopLimit > 0, error = kThreadError_Drop);
+        VerifyOrExit(hopLimit-- > 1, error = kThreadError_Drop);
```

Comment:
Theoretically, someone (malicious) could send a packet with a hop limit of 0. Decrementing first before doing the comparison, would result in the code not dropping the packet.